api/envoy/config/route/v3/route_components.proto

```diff
@@ -1929,6 +1929,10 @@ message HeaderMatcher {
   // * The regex ``\d{3}`` does not match the value *1234*, so it will match when inverted.
   // * The range [-10,0) will match the value -1, so it will not match when inverted.
   bool invert_match = 8;
+
+  // If true, indicates the exact_match/prefix_match/suffix_match/contains_match should be case
+  // insensitive. This has no effect for the safe_regex_match.
+  bool ignore_case = 13;
```

Comment:
Instead of adding this here, I think it would make more sense to add a [`type.matcher.v3.StringMatcher`](https://github.com/envoyproxy/envoy/blob/dfef049fe9692ff8188a83bab859b310c212ece1/api/envoy/type/matcher/v3/string.proto#L20) field and deprecate the existing `exact_match`, `safe_regex_match`, `prefix_match`, `suffix_match`, and `contains_match` fields.  That way, we can eventually get away from having duplicate implementations here.

Note that `StringMatcher` already supports an `ignore_case` field.

Reply:
Sure I'm fine with StringMatcher, it is a slightly larger API change and deprecation. Also cc @htuch in case if there are any other comments about this.

Reply:
I think we discussed this in the past and there was a concern we might break a lot of exiting users via a deprecation here. OTOH, since we don't have major versions coming up, I think we need some way to pay down API tech debt. I'm good with this if other API shepherds agree, but would like to check with @mattklein123 who had the reservation last time this came up.

Reply:
Yeah I agree this is going to be a very large change for everyone, and kind of annoying, but as @htuch says if v4 is "never" we need to go ahead and do stuff like this. Given that now we are talking about 2 year+ cycles for deprecation removal I think it's fine to go ahead and do it.

Reply:
@htuch @mattklein123 thanks for the comment, the PR is ready for review, PTAL.
I updated again and all tests are passing except the Mac CI test timed out after 3 hours which seems not related to the PR.